crates/cards/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+use common_utils::{date_time, errors};
+use error_stack::report;
+use masking::{PeekInterface, StrongSecret};
+use time::{util::days_in_year_month, Date, Duration, PrimitiveDateTime, Time};
+
+pub struct CardSecurityCode(StrongSecret<u16>);
+
+impl CardSecurityCode {
+    pub fn new(secret: StrongSecret<u16>) -> errors::CustomResult<Self, errors::ValidationError> {
+        let csc = secret.peek();
+
+        if *csc > 99 && *csc < 10000 {
+            Ok(Self(secret))
+        } else {
+            Err(report!(errors::ValidationError::InvalidValue {
+                message: "invalid card security code".to_string()
+            }))
+        }
+    }
+}
+
+pub struct CardExpirationMonth(StrongSecret<u8>);
+
+impl CardExpirationMonth {
+    pub fn new(secret: StrongSecret<u8>) -> errors::CustomResult<Self, errors::ValidationError> {
+        let month = secret.peek();
+
+        if *month >= 1 && *month <= 12 {
+            Ok(Self(secret))
+        } else {
+            Err(report!(errors::ValidationError::InvalidValue {
+                message: "invalid card expiration month".to_string()
+            }))
+        }
+    }
+
+    pub fn two_digits(&self) -> String {
+        let month = self.0.peek();
+        let month_str: String = month.to_string();
+        if month_str.len() == 1 {
+            format!("0{}", month_str)
+        } else {
+            month_str
+        }
+    }
+}
+
+pub struct CardExpirationYear(StrongSecret<u16>);
+
+impl CardExpirationYear {
+    pub fn new(secret: StrongSecret<u16>) -> errors::CustomResult<Self, errors::ValidationError> {
+        let year = secret.peek();
+
+        if *year >= 1997 {
```

Comment:
Is there a specific reason why this check is done against 1997? Won't it be better to check if it's `>= current_year`?

Reply:
sure, I'll change that